WordPress/Classes/ViewRelated/Reader/ReaderPostCardCell.swift

```diff
@@ -0,0 +1,580 @@
+        } else {
+            avatarImageView.image = placeholder
+        }
+
+        var blogName = contentProvider?.blogNameForDisplay()
+        blogNameButton.setTitle(blogName, forState: .Normal)
+        blogNameButton.setTitle(blogName, forState: .Highlighted)
+
+        var byline = contentProvider?.dateForDisplay().shortString()
+        if let author = contentProvider?.authorForDisplay() {
+            byline = String(format: "%@, %@", byline!, author)
+        }
+
+        bylineLabel.text = byline
+    }
+
+    private func configureCardImage() {
+        // Always clear the previous image so there is no stale or unexpected image 
+        // momentarily visible.
+        featuredImageView.image = nil
+        if let featuredImageURL = contentProvider?.featuredImageURLForDisplay?() {
+            if loadMediaWhenConfigured {
+                var url = featuredImageURL
+                if !(contentProvider!.isPrivate()) {
+                    let size = CGSize(width:featuredMediaView.frame.width, height:featuredMediaHeightConstraintConstant)
+                    url = PhotonImageURLHelper.photonURLWithSize(size, forImageURL: url)
+                    featuredImageView.setImageWithURL(url, placeholderImage:nil)
+
+                } else if (url.host != nil) && url.host!.hasSuffix("wordpress.com") {
+                    // private wpcom image needs special handling. 
+                    let request = requestForURL(url)
+                    featuredImageView.setImageWithURLRequest(request, placeholderImage: nil, success: nil, failure: nil)
+
+                } else {
+                    // private but not a wpcom hosted image
+                    featuredImageView.setImageWithURL(url, placeholderImage:nil)
+                }
+            }
+            featuredMediaHeightConstraint.constant = featuredMediaHeightConstraintConstant
+            featuredMediaBottomConstraint.constant = featuredMediaBottomConstraintConstant
+
+        } else {
+            featuredMediaHeightConstraint.constant = 0.0
+            featuredMediaBottomConstraint.constant = 0.0
+        }
+
+        featuredMediaView.setNeedsUpdateConstraints()
+    }
+
+    private func requestForURL(url:NSURL) -> NSURLRequest {
+        var requestURL = url
+        if let absoluteString = requestURL.absoluteString {
+            if !(absoluteString.hasPrefix("https")) {
+                var sslURL = absoluteString.stringByReplacingOccurrencesOfString("http", withString: "https")
+                requestURL = NSURL(string: sslURL)!
+            }
+        }
+
+        let acctServ = AccountService(managedObjectContext: ContextManager.sharedInstance().mainContext)
+        let token = acctServ.defaultWordPressComAccount().authToken
+        var request = NSMutableURLRequest(URL: requestURL)
+        var headerValue = String(format: "Bearer %@", token)
+        request.addValue(headerValue, forHTTPHeaderField: "Authorization")
```

Comment:
I looked.  `requestForAuthenticationWithURL:redirectURL:username:password:bearerToken:userAgent:` isn't a good fit here. `loginUrl`, `redirectURL`, and `username` are required.  
On the other hand, we could add a new convenience method for adding the token to a request, I'm just not sure we'd gain much since the caller would probably still need to obtain and pass the token. (I don't think it would make sense for `WPURLRequest` to do that itself.) But maybe its worth it if there are other places we need to add the token to a request.